g2p/mappings/__init__.py

```diff
@@ -176,34 +180,36 @@ def process_model_specs(self):  # noqa: C901
             self.rules = sorted(
                 # Temporarily normalize to NFD for heuristic sorting of NFC-defined rules
                 self.rules,
-                key=lambda x: (
-                    len(normalize(strip_index_notation(x.rule_input), "NFD"))
-                    if isinstance(x, Rule)
-                    else len(normalize(x["in"], "NFD"))
-                ),
```

Comment:
Interestingly, removing this conditional also fixed a bug that has `umista_equiv.csv` sorted before removing its index notation instead of after.
Presumably the real different here is that you're creating `Rule` objects right away all the time, rather initializing the `Mapping` class with either/or.

Reply:
oh - hopefully I haven't broken some subtle behaviour that we depended on, it seemed to make sense to me to create the `Rule` objects right away, so that they always have the same validation